README.md

```diff
@@ -461,6 +461,73 @@ TL;DR for a simple backup of three PostgreSQL databases in the same project name
+
+
+TL;DR: A simple backup of a single MongoDB database with backup validation.
+
+1. Decide on amount of backup storage required (5Gi is currently the maximum)
+2. Provision the nfs-backup PVC, following the [docs](https://github.com/BCDevOps/provision-nfs-apb/blob/master/docs/usage-gui.md). This provisioning may take several minutes to an hour, and if using the GUI, will result in a PVC with a name similar to `bk-abc123-dev-v9k7xgyvwdxm`, where `abc123-dev` is your project namespace and the last portion is randomly generated.
+3. `git clone https://github.com/BCDevOps/backup-container.git && cd backup-container`.
+4. Determine the OpenShift namespace for the image (e.g. `abc123-dev`), the app name (e.g. `myapp-backup`), and the image tag (e.g. `v1`). Then build the image in your the namespace.
+```bash
+oc -n abc123-dev process -f ./openshift/templates/backup/backup-build.json \
+  -p DOCKER_FILE_PATH=Dockerfile_Mongo
+  -p NAME=myapp-backup OUTPUT_IMAGE_TAG=v1 | oc -n abc123-dev create -f -
+```
+5. Configure `./config/backup.conf`. This defines the database(s) to backup and the schedule that backups are to follow. Additionally, this sets up backup validation (identified by `-v all` flag).
+```bash
+# Database(s)
+mongo=myapp-mongodb:27017/mydb
+
+# Cron Schedule(s)
+0 1 * * * default ./backup.sh -s
+0 4 * * * default ./backup.sh -s -v all
+```
+6. Configure references to your DB credentials in [backup-deploy.json](./openshift/templates/backup/backup-deploy.json), replacing the boilerplate `DATABASE_USER` and `DATABASE_PASSWORD` environment variable names. Note the hostname of the database to be backed up. This example uses a hostname of `myapp-mongodb` which maps to environement variables named `MYAPP_MONGODB_USER` and `MYAPP_MONGODB_PASSWORD`. See the [backup.conf](#backupconf) section  above for more in depth instructions. This example also assumes that the name of the secret containing your database username and password is the same as the provided `DATABASE_DEPLOYMENT_NAME` parameter. If that's not the case for your service, the secret name can be overridden.
+```json
+{
+  "name": "MYAPP_MONGODB_USER",
+  "valueFrom": {
+    "secretKeyRef": {
+      "name": "${DATABASE_DEPLOYMENT_NAME}",
+      "key": "${DATABASE_USER_KEY_NAME}"
+    }
+  }
+},
+{
+  "name": "MYAPP_MONGODB_PASSWORD",
+  "valueFrom": {
+    "secretKeyRef": {
+      "name": "${DATABASE_DEPLOYMENT_NAME}",
+      "key": "${DATABASE_PASSWORD_KEY_NAME}"
+    }
+  }
+},
+```
+8. Deploy the app. In this example, the namespace is `abc123-dev` and the app name is `myapp-backup`. Note that the key names within the database secret referencing database username and password are `username` and `password`, respectively. If this is not the case for your deployment, specify the correct key names as parameters `DATABASE_USER_KEY_NAME` and `DATABASE_PASSWORD_KEY_NAME`. Also note that `BACKUP_VOLUME_NAME` is from Step 2 above.
+
+```bash
```

Comment:
Alternatively to all of the discrete `oc` commands you can integrate the backup configurations (Build and Deployment templates, override script, and config) directly into your project configuration and manage the publishing and updating of the Build and Deployment configurations using the [BCDevOps/openshift-developer-tools ](https://github.com/BCDevOps/openshift-developer-tools/tree/master/bin) scripts.  An example can be found in the [bcgov/orgbook-configurations](https://github.com/bcgov/orgbook-configurations), [backup](https://github.com/bcgov/orgbook-configurations/tree/master/openshift/templates/backup) templates folder.

Reply:
Simplified documentation on how to use the tools can be found here; https://github.com/bcgov/jag-cullencommission/tree/master/openshift.  All scripts support the `-c` option with allows you to perform operations on a single `component` of you application such as the backup container (`-c backup`, in the `orgbook-configurations` example above).

Reply:
This is super helpful and certainly merits adding to the README. I'm not sure if it belongs in a specific example of deploying a backup container for MongoDB, however. I'm happy to add this, but wanted your thoughts on where it best fits. Maybe I could add your comments above to the Backups in OpenShift section.

Reply:
Yes, perhaps add the additional details to the https://github.com/BCDevOps/backup-container/blob/a3b13823b51171d39bb8da4343f861d4800db07b/README.md#backups-in-openshift section.  I see the tools are already mentioned there but I find most teams seem to miss the references.

Reply:
Done.